### PR TITLE
refactor(middleware): update middleware/common to ES6

### PR DESCRIPTION
**Pull request by johnjbarton**

### CODE REVIEW

File: lib/middleware/common.js [hunk 1]
```diff
@@ -117,7 +120,7 @@ var createServeFile = function (fs, directory, config) {
       responseData = (transform && transform(data.toString())) || data
 
       if (rangeHeader) {
-        var code = convertForRangeRequest()
+        const code = convertForRangeRequest()
```
Comment (lusarz) [inline lib/middleware/common.js]: You can update L122-L127 into `response.writeHead(rangeHeader ? convertForRangeRequest() : 200)`

File: lib/middleware/common.js [hunk 2]
```diff
@@ -1,41 +1,44 @@
...
-var serve404 = function (response, path) {
+const serve404 = function (response, path) {
   log.warn('404: ' + path)
   response.writeHead(404)
   return response.end('NOT FOUND')
 }
 
-var createServeFile = function (fs, directory, config) {
-  var cache = Object.create(null)
+const createServeFile = function (fs, directory, config) {
```
Comment (lusarz) [inline lib/middleware/common.js]: What do you think about `function createServeFile (fs, directory, config)` instead of `const` ?

Comment (johnjbarton) [reply lib/middleware/common.js]: done (also others)

File: lib/middleware/common.js (hunk 1 above)
Comment (johnjbarton) [reply lib/middleware/common.js]: done

**Pull request merged**